packages/nextjs-demo/src/app/docs-chat/page.tsx

```diff
@@ -0,0 +1,84 @@
+/** @jsxImportSource ai-jsx/react */
+import * as AI from 'ai-jsx/next';
+import * as React from 'react';
+import { Suspense } from 'react';
+import { DocsAgent } from './ai';
+import ResultContainer from '@/components/ResultContainer';
+
+function ConversationItem({
+  responseType,
+  children: responseContent,
+}: {
+  responseType: 'user' | 'bot';
+  children: React.ReactNode;
+}) {
+  const emoji = responseType === 'user' ? '👤' : '🤖';
+  return (
+    <div>
+      {emoji}: {responseContent}
+    </div>
+  );
+}
+
+/**
+ * We need to memoize this function. Otherwise, every time the parent component of AgentResponse re-renders,
+ * AgentReponse will re-render, which will trigger a new LLM call, which will produce a different result than the
+ * prior call.
+ */
+const AgentResponse = React.memo(function AgentResponse({ question }: { question: string }) {
+  return (
+    <ConversationItem responseType="bot">
+      <Suspense fallback="⎕">
+        <AI.jsx>
+          <DocsAgent question={question} />
+        </AI.jsx>
+      </Suspense>
+    </ConversationItem>
+  );
+});
+
+export default function DocsChat({ searchParams }: { searchParams: any }) {
+  const defaultValue = 'What is AI.JSX?';
+  const query = searchParams.message ?? defaultValue;
+  const userMessages: string[] = [query];
```

Comment:
```suggestion
  const userMessages = [query];
```

Reply:
I don't think this type annotation is necessary. We want to let type inference do as much as it can.